File: NeetCode/13-Intervals/MergeIntervals.java
import java.util.*;

public class MergeIntervals {
    public int[][] merge(int[][] intervals) {
        
        Arrays.sort(intervals, (a, b) -> a[0] - b[0]);
        List<int[]> result = new ArrayList<>();
        result.add(intervals[0]);
        for(int i = 1; i < intervals.length; i++){
            int[] getLast = result.get(result.size() - 1);
            if(intervals[i][0] <= getLast[1]){
                getLast[1] = Math.max(intervals[i][1], getLast[1]);
            }
            else{
                result.add(intervals[i]);
            }
        }
        return result.toArray(new int[result.size()][]);
    }
}

File: NeetCode/13-Intervals/MergeIntervals.cpp
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end());
        vector<vector<int>> result;
        result.push_back(intervals[0]);
        for(int i = 1; i < intervals.size(); i++){
            vector<int>& getLast = result.back();
            if(intervals[i][0] <= getLast[1]){
                getLast[1] = max(intervals[i][1], getLast[1]);
            }
            else{
                result.push_back(intervals[i]);
            }
        }
        return result;
    }
};